lib/spack/spack/binary_distribution.py

```diff
@@ -359,7 +361,8 @@ def relocate_package(prefix):
     path_names = set()
     for filename in buildinfo['relocate_textfiles']:
         path_name = os.path.join(prefix, filename)
-        path_names.add(path_name)
+        if '~' not in path_name:
```

Comment:
Is this related to:

> files ending in ~ are removed by filter_file(backup=False) but the files might end up in the text to relocate list is filter_file() is called during install

If so, IMO it would be clearer to use `endswith` because otherwise I was confused if you were looking for paths starting with `~` (i.e. referring to the home directory).

If you are wanting to remove paths which end with `~` (which I think is a backup file created by some editors) then IMO that isn't something you should have to deal with. If you think it ought to then IMO it is worth adding a comment about why this check is there.

Reply:
I found one case where there was a file ending in ~ because filter_file was called during install. The file ended up on the list of text relocation files. It was overwritten when the parent file was changed and then removed.

Reply:
Got it - `filter_file` itself generates these. Could you replace this with a call to `endswith` and add a comment about how this comes from `filter_file`?